src/util/Parameters.h

```diff
@@ -31,6 +34,26 @@ struct ParameterBase {
   virtual ~ParameterBase() = default;
 };
 
+// Concepts for the template types of `Parameter`.
+template <typename T>
+concept ParameterValueType =
+    std::same_as<std::decay_t<T>, T> && std::default_initializable<T> &&
+    std::movable<T> && std::copyable<T>;
```

Comment:
Check (and experiment) that you can just use `std::semiregular` instead of this concept
(References aren't default_initializable) etc.

Reply:
I could use `std::semiregular`, no problem. But I don't really see, **why**. It would weaken the concept and not describe all the things, that were required of the type. For example, that it is movable.

> References aren't default_initializable

I don't follow. The code of `Parameter` only works with types, that **are** default initializable. Additionally, `std::semiregular` also requires it, so it wouldn't change anything.

Reply:
Okay, `std::copyable` implies `std::movable` as by my short tests.
And `semiregular` implies `copyable`  and `default_initializable`.
And since references, and const /volatile types are not `copyable`,
Your manual constraints are a subset (or maybe even equivalent) to the stanrd concept `std::semiregular`, so you should use that one and not reimplement it. It does **not** weaken the concept.

Reply:
> Okay, `std::copyable` implies `std::movable` as by my short tests. 

Ah, by bad. I didn't think of looking up, if there was overlap between the concepts and there exact definition.